plugins/woocommerce-admin/client/marketplace/components/product-card/product-card.tsx

```diff
@@ -21,10 +22,19 @@ function ProductCard( props: ProductCardProps ): JSX.Element {
 	// We hardcode this for now while we only display prices in USD.
 	const currencySymbol = '$';
 
+	let productUrl = '';
+	if ( product.url ) {
+		const wccomHelperSettings = getAdminSetting( 'wccomHelper', {} );
```

Comment:
We'll be running this `getAdminSetting` for each `ProductCard` on the page. I wonder if it'll make things slower. Alternatively, we can move this logic to the `ProductListContextProvider` so that we'll run the `getAdminSetting` once and change all product URLs according to that. 

I'd be happy to make the change but wanted to see what do you think.

Reply:
Please go ahead! This was just the bare bones prototype to get things started.

Reply:
Ah, I found a problem in our approach with using the context. Unfortunately, since Discover doesn't yet utilise the context provider, we're unable to get this data from context on that page, making the issue still exist.

I do think it's the right approach eventually, but once we've moved Discover to using the context first. 

Until then, I suspect the better approach would be to take a similar, but less ideal approach and run the `getAdminSetting()` call in the `productList` component, so it's still only handled once. We can then pass the data into the cards from there.

Reply:
Alright, thanks for thinking this through! I added a bit of logic to the `ProductListContent` component. Same logic could also be implemented to the `ProductContent` though, I think not much difference there. 

That allowed me to keep the `ProductCard` as simple as possible. No more WCCOM logic in that component. It just prints whatever URL is passed to it.